### PR TITLE
Use platform dependency in gradle files

### PR DESCRIPTION
**Pull request by dupliaka**
Suggestion to move to platform resolution of optaplanner-bom
If we decide to switch to the platform I will add another commit to update Spring and Quarkus example as well

### CODE REVIEW

**Review by ge0ffrey — APPROVED** [review]

**Comment by ge0ffrey** [comment]
@dupliaka Great improvement. Can we do that for the other 2 build.gradle files too? (quarkus and spring school timetabling)

**Comment by dupliaka** [comment]
> @dupliaka Great improvement. Can we do that for the other 2 build.gradle files too? (quarkus and spring school timetabling)

Do we need it there? I thought that if you create your application based on Quarkus platform then you use those Quarkus dependencies.

And only if you create plain java application/optaplanner-service then you would need to use optaplanner platform

**Comment by ge0ffrey** [comment]
We do need them there for both quarkus and spring, because we can't depend on the quarkus platform bom due to chicken-and-egg problems: the quickstarts development branch must use features only available in an optaplanner snapshot (not yet released features), so it can by definition not use the quarkus platform bom that uses an optaplanner release.

The quarkus school timetabling quickstarts depends on optaplanner-test too.

**Review by rsynek — APPROVED** [review]

**Comment by dupliaka** [comment]
> We do need them there for both quarkus and spring, because we can't depend on the quarkus platform bom due to chicken-and-egg problems: the quickstarts development branch must use features only available in an optaplanner snapshot (not yet released features), so it can by definition not use the quarkus platform bom that uses an optaplanner release.
> 
> The quarkus school timetabling quickstarts depends on optaplanner-test too.

Ok, I updated other gradle files, please take a look

File: technology/java-spring-boot/build.gradle [hunk 1]
```diff
@@ -22,20 +22,21 @@ repositories {
 }
 
 dependencies {
+    implementation enforcedPlatform("org.optaplanner:optaplanner-bom:${optaplannerVersion}")
```
Comment (ge0ffrey) [inline technology/java-spring-boot/build.gradle]: optaplanner should be after spring-boot

Motivation: optaplanner is not the platform, spring is.
OptaPlanner is just one of many libraries/extensions, it cannot play dependency management boss.
if spring-boot defines slf4j 1.2 and we define another slf4j version, then the platform version must win.
optaplanner must be able to adjust to the platform, not visa versa.

File: use-cases/school-timetabling/build.gradle [hunk 2]
```diff
@@ -21,12 +21,12 @@ repositories {
...
 
 dependencies {
-    // Alternatively, use "io.quarkus:quarkus-universe-bom" which includes both quarkus-bom and optaplanner-bom.
-    implementation enforcedPlatform("io.quarkus:quarkus-bom:${quarkusVersion}")
-    implementation "org.optaplanner:optaplanner-quarkus:${optaplannerVersion}"
-    implementation "org.optaplanner:optaplanner-quarkus-jackson:${optaplannerVersion}"
-    implementation "org.optaplanner:optaplanner-quarkus-benchmark:${optaplannerVersion}"
-    implementation "org.optaplanner:optaplanner-test:${optaplannerVersion}"
+    implementation enforcedPlatform("org.optaplanner:optaplanner-bom:${optaplannerVersion}")
+    implementation platform("io.quarkus:quarkus-bom:${quarkusVersion}")
```
Comment (ge0ffrey) [inline use-cases/school-timetabling/build.gradle:24]: optaplanner should be after quarkus

Motivation: optaplanner is not the platform, quarkus is.
OptaPlanner is just one of many libraries/extensions, it cannot play dependency management boss.
if quarkusdefines slf4j 1.2 and we define another slf4j version, then the platform version must win.
optaplanner must be able to adjust to the platform, not visa versa.

**Comment by ge0ffrey** [comment]
@dupliaka In both cases, can you move the optaplanner-bom line after the quarkus/spring lines?
If that's fixed, this is good to merge. Good work :)

**Comment by ge0ffrey** [comment]
Thanks Anna!

**Pull request merged**